File: gitlfs-client/src/main/java/ru/bozaro/gitlfs/client/io/ByteArrayStreamProvider.java
package ru.bozaro.gitlfs.client.io;

import javax.annotation.Nonnull;
import java.io.ByteArrayInputStream;
import java.io.InputStream;

/**
 * Create stream by bytes.
 *
 * @author Artem V. Navrotskiy
 */
public class ByteArrayStreamProvider implements StreamProvider {
  @Nonnull
  private final byte[] data;

  public ByteArrayStreamProvider(@Nonnull byte[] data) {
    this.data = data;
  }

  @Nonnull
  @Override
  public InputStream getStream() {
    return new ByteArrayInputStream(data);
  }
}

File: gitlfs-client/src/main/kotlin/ru/bozaro/gitlfs/client/io/ByteArrayStreamProvider.kt
package ru.bozaro.gitlfs.client.io

import java.io.ByteArrayInputStream
import java.io.InputStream

/**
 * Create stream by bytes.
 *
 * @author Artem V. Navrotskiy
 */
open class ByteArrayStreamProvider(private val data: ByteArray) : StreamProvider {
    override val stream: InputStream
        get() = ByteArrayInputStream(data)
}
